include/libopencm3/stm32/h7/rcc.h

```diff
@@ -181,23 +328,228 @@ LGPL License Terms @ref lgpl_license
+};
+
+enum rcc_d1hpre {
+  RCC_D1HPRE_BYP = RCC_D1CFGR_D1HPRE_BYP,
+  RCC_D1HPRE_DIV2 = RCC_D1CFGR_D1HPRE_DIV2,
+  RCC_D1HPRE_DIV4 = RCC_D1CFGR_D1HPRE_DIV4,
+  RCC_D1HPRE_DIV8 = RCC_D1CFGR_D1HPRE_DIV8,
+  RCC_D1HPRE_DIV16 = RCC_D1CFGR_D1HPRE_DIV16,
+  RCC_D1HPRE_DIV64 = RCC_D1CFGR_D1HPRE_DIV64,
+  RCC_D1HPRE_DIV128 = RCC_D1CFGR_D1HPRE_DIV128,
+  RCC_D1HPRE_DIV256 = RCC_D1CFGR_D1HPRE_DIV256,
+  RCC_D1HPRE_DIV512 = RCC_D1CFGR_D1HPRE_DIV512,
+};
+
+enum rcc_d2ppre1 {
+  RCC_D2PPRE1_BYP = RCC_D2CFGR_D2PPRE_BYP << RCC_D2CFGR_D2PPRE1_SHIFT,
+  RCC_D2PPRE1_DIV2 = RCC_D2CFGR_D2PPRE_DIV2 << RCC_D2CFGR_D2PPRE1_SHIFT,
+  RCC_D2PPRE1_DIV4 = RCC_D2CFGR_D2PPRE_DIV4 << RCC_D2CFGR_D2PPRE1_SHIFT,
+  RCC_D2PPRE1_DIV8 = RCC_D2CFGR_D2PPRE_DIV8 << RCC_D2CFGR_D2PPRE1_SHIFT,
+  RCC_D2PPRE1_DIV16 = RCC_D2CFGR_D2PPRE_DIV16 << RCC_D2CFGR_D2PPRE1_SHIFT,
+};
+
+enum rcc_d2ppre2 {
+  RCC_D2PPRE2_BYP = RCC_D2CFGR_D2PPRE_BYP << RCC_D2CFGR_D2PPRE2_SHIFT,
+  RCC_D2PPRE2_DIV2 = RCC_D2CFGR_D2PPRE_DIV2 << RCC_D2CFGR_D2PPRE2_SHIFT,
+  RCC_D2PPRE2_DIV4 = RCC_D2CFGR_D2PPRE_DIV4 << RCC_D2CFGR_D2PPRE2_SHIFT,
+  RCC_D2PPRE2_DIV8 = RCC_D2CFGR_D2PPRE_DIV8 << RCC_D2CFGR_D2PPRE2_SHIFT,
+  RCC_D2PPRE2_DIV16 = RCC_D2CFGR_D2PPRE_DIV16 << RCC_D2CFGR_D2PPRE2_SHIFT,
+};
+
+enum rcc_d3ppre {
+  RCC_D3PPRE_BYP = RCC_D3CFGR_D3PPRE_BYP << RCC_D3CFGR_D3PPRE_SHIFT,
+  RCC_D3PPRE_DIV2 = RCC_D3CFGR_D3PPRE_DIV2 << RCC_D3CFGR_D3PPRE_SHIFT,
+  RCC_D3PPRE_DIV4 = RCC_D3CFGR_D3PPRE_DIV4 << RCC_D3CFGR_D3PPRE_SHIFT,
+  RCC_D3PPRE_DIV8 = RCC_D3CFGR_D3PPRE_DIV8 << RCC_D3CFGR_D3PPRE_SHIFT,
+  RCC_D3PPRE_DIV16 = RCC_D3CFGR_D3PPRE_DIV16 << RCC_D3CFGR_D3PPRE_SHIFT,
+};
+
+/** PLL Configuration structure. */
+struct rcc_pll_config {
+  uint32_t hse_frequency;         /**< User configured external crystal frequency. */
+  enum rcc_sysclk_mux sysclk_mux; /**< SYSCLK source input selection. */
+  enum rcc_pll_mux pll_mux;       /**< PLL source input selection. */
+  struct {
+    uint8_t divm;                 /**< Pre-divider value for each PLL. 0-64 integers. */
+    uint16_t divn;                /**< Multiplier, 0-512 integer. */
+    uint8_t divp;                 /**< Post divider for PLLP clock. */
+    uint8_t divq;                 /**< Post divider for PLLQ clock. */
+    uint8_t divr;                 /**< Post divider for PLLR clock. */
+  } pll[3];                      /**< Array of 3x PLLs. */
+  struct {
+    enum rcc_d1cpre core_prescale;    /**< Core prescaler for domain 1. */
+    enum rcc_d1hpre hclk3_prescale;   /**< HCLK3 prescaler for domain 1. */
+    enum rcc_d1ppre pclk3_prescale;   /**< APB3 Peripheral prescaler for domain 1. */
+  } domain1;
+  struct {
+    enum rcc_d2ppre1 pclk1_prescale;  /**< APB1 Peripheral prescaler for domain 2. */
+    enum rcc_d2ppre2 pclk2_prescale;  /**< APB2 Peripheral prescaler for domain 2. */
+  } domain2;
+  struct {
+    enum rcc_d3ppre pclk4_prescale;   /**< APB4 Peripheral prescaler for domain 3. */
+  } domain3;
```

Comment:
unless you're expecting people to pass around and work withi the inner structs individually, just make it flat, there's no reason for the extra layers.

Reply:
flattened this out and added prefix for the register for documentation.